wasi-sdk-pthread.cmake

```diff
@@ -1,8 +1,5 @@
 # Cmake toolchain description file for the Makefile
 
-# This is arbitrary, AFAIK, for now.
-cmake_minimum_required(VERSION 3.4.0)
```

Comment:
The android NDK toolchain file does seem to include a `cmake_minimum_version`: https://android.googlesource.com/platform/ndk/+/master/build/cmake/android.toolchain.cmake#35

Are you sure this is not supposed to be here?

Perhaps we could bump to to 3.6 like the android one to avoid the warning you mention?

Reply:
Yes, it is not supposed to be there. Was it originally included for a specific reason?

See the [examples in the official CMake documentation](https://github.com/Kitware/CMake/blob/e449417f9468ffada6c0e38a5dab09122a3d1c25/Help/manual/cmake-toolchains.7.rst#toolchain-features). Or [these examples](https://github.com/dockcross/dockcross).

The warning I pointed to is one example of unintended consequences, but there are more. CMake changes its behavior [based on policies](https://cmake.org/cmake/help/latest/command/cmake_policy.html), which are impacted by a `cmake_minimum_required` call.

Reply:
That line as been there since the initial commit (c80964d8d7cf8ca2480d8332cd8022cb0600d421).  

I don't know it was included, it looks like we would have to go back to `yury's wasmception` to find out why.

It sounds like you know what you are talking about cmake-wise, and also emscripten's cmake does not specify `cmake_minimum_required` so I'm tempted to just approve this.

Lets wait for @pchickey to chime in though.

Reply:
Thank you for your attention and consideration.

Reply:
I agree that this change is likely the right one to make.